[FSDP][2/N] Remove `params_with_grad`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#87480 [FSDP][2/N] Remove `params_with_grad`**
* #87479 [FSDP][1/N] Rework `clip_grad_norm_()` and tests
* #87478 [FSDP][Docs] Clarify warnings to mention collectives

This PR removes the property `params_with_grad` from `FullyShardedDataParallel`. It was introduced when implementing `clip_grad_norm_()` but was not consistently used. Personally, I do not think it makes sense for `FullyShardedDataParallel` to expose this helper because it is not a common paradigm.

This PR is technically BC-breaking. However, I checked that no one internally is using this API.


cc @ezyang @gchanan

Commit:
[FSDP] Remove `params_with_grad`

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Remove `params_with_grad`"


This PR removes the property `params_with_grad` from `FullyShardedDataParallel`. It was introduced when implementing `clip_grad_norm_()` but was not consistently used. Personally, I do not think it makes sense for `FullyShardedDataParallel` to expose this helper because it is not a common paradigm.

This PR is technically BC-breaking. However, I checked that no one internally is using this API.


[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Remove `params_with_grad`"


This PR removes the property `params_with_grad` from `FullyShardedDataParallel`. It was introduced when implementing `clip_grad_norm_()` but was not consistently used. Personally, I do not think it makes sense for `FullyShardedDataParallel` to expose this helper because it is not a common paradigm.

This PR is technically BC-breaking. However, I checked that no one internally is using this API.


[ghstack-poisoned]

rohan-varma reviewed (approved):
Sounds good please add BC breaking label for release tracking purposes.

Commit:
Update on "[FSDP][2/N] Remove `params_with_grad`"


This PR removes the property `params_with_grad` from `FullyShardedDataParallel`. It was introduced when implementing `clip_grad_norm_()` but was not consistently used. Personally, I do not think it makes sense for `FullyShardedDataParallel` to expose this helper because it is not a common paradigm.

This PR is technically BC-breaking. However, I checked that no one internally is using this API.


cc @ezyang @gchanan

[ghstack-poisoned]

awgu commented:
@pytorchbot merge
